networkx/algorithms/operators/binary.py

```diff
@@ -324,8 +324,8 @@ def compose(G, H, name=None):
         name = "compose( %s, %s )" % (G.name, H.name)
     R = G.__class__()
     R.name = name
-    R.add_nodes_from(H.nodes())
-    R.add_nodes_from(G.nodes())
+    R.add_nodes_from(G.nodes_iter(data=True))
```

Comment:
Could you use G.nodes() instead of G.nodes_iter() please?  In the next couple of days we hope to merge a change that makes the nodes() methods iterators. It will make that process slightly easier if you use nodes() here.
Thanks!